keys/keys.go

```diff
@@ -471,16 +471,27 @@ func MakeSplitKey(key roachpb.Key) (roachpb.Key, error) {
 // Range returns a key range encompassing all the keys in the Batch.
 // TODO(tschottdorf): there is no protection for doubly-local keys here;
 // maybe Range should return an error.
-func Range(ba roachpb.BatchRequest) roachpb.RSpan {
+func Range(ba roachpb.BatchRequest) (roachpb.RSpan, error) {
 	from := roachpb.RKeyMax
 	to := roachpb.RKeyMin
-	for _, arg := range ba.Requests {
+	for i, arg := range ba.Requests {
 		req := arg.GetInner()
 		if req.Method() == roachpb.Noop {
 			continue
 		}
 		h := req.Header()
 		key := Addr(h.Key)
+		if i == len(ba.Requests) {
+			if bytes.Compare(h.Key, from) >= 0 {
+				return roachpb.RSpan{Key: from, EndKey: to}, util.Errorf("%s: must be less than KeyMax", h.Key)
```

Comment:
on error, usually you return the zero values for all of the other return values, so `return nil, util.Errorf(...)`.